app/models/manageiq/providers/hawkular/datawarehouse_manager.rb

```diff
@@ -33,9 +34,18 @@ def validate_authentication_status
       {:available => true, :message => nil}
     end
 
-    def self.verify_ssl_mode
-      # TODO: support real authentication using certificates
-      OpenSSL::SSL::VERIFY_NONE
+    def verify_ssl_mode(endpoint = default_endpoint)
+      case endpoint.security_protocol
+      when 'ssl-without-validation'
+        OpenSSL::SSL::VERIFY_NONE
+      else # 'ssl-with-validation', 'ssl-with-validation-custom-ca', secure by default with unexpected values.
+        OpenSSL::SSL::VERIFY_PEER
+      end
+    end
+
+    def ssl_cert_store(endpoint = default_endpoint)
+      # Given missing (nil) endpoint, return nil meaning use system CA bundle
+      endpoint.try(:ssl_cert_store)
```

Comment:
Containers provider needed this `.try` for the 'hawkular' endpoint which was optional there.
If you have only one always present default endpoint, you may drop the `try` here or even inline `default_endpoint.ssl_cert_store` in `connect`.  Your call.